### PR TITLE
feat(packages/appeal-reply-service-api): initial commit

### PR DESCRIPTION
**Pull request by swdpcomputing**
## Ticket Number

AS-54

## Checklist

- [x] Requires infrastructure changes
- [ ] If adding or remove environment variables (e.g. in `docker-compose.yaml`) then I have updated the appropriate Helm chart
- [ ] I have updated the documentation accordingly
- [x] My commit history in this PR is linear
- [x] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `master` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: packages/appeal-reply-service-api/api/openapi.yaml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+info:
+  title: Appeal Reply Service API Documentation
+  description:
+    API to <ul><li>create, retrieve and modify appeal replies</li><li>
+  version: 1.0.3
+  license:
+    name: MIT
+    url: 'https://opensource.org/licenses/MIT'
+servers:
+  - url: 'http://localhost:3000/'
```
Comment (MrSimonEmms) [inline packages/appeal-reply-service-api/api/openapi.yaml:11]: This _probably_ wants to be changed to the port assigned in Docker Compose

File: packages/appeal-reply-service-api/data/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Data
```
Comment (MrSimonEmms) [inline packages/appeal-reply-service-api/data/README.md:1]: The `/data` folder is not required - remove

File: packages/appeal-reply-service-api/data/lpa-trialists.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+[
```
Comment (MrSimonEmms) [inline packages/appeal-reply-service-api/data/lpa-trialists.json:1]: Not required

File: packages/appeal-reply-service-api/globalConfig.json [hunk 4]
```diff
@@ -0,0 +1 @@
+{"mongoUri":"mongodb://127.0.0.1:52242/jest?","mongoDBName":"jest"}
```
Comment (MrSimonEmms) [inline packages/appeal-reply-service-api/globalConfig.json:1]: I've worked out why this is a problem in this package, but not in appeals-service-api.

In `/.gitignore`, please change:
```
// line ~171
packages/appeals-service-api/globalConfig.json
```

to:
```
packages/**/globalConfig.json
```

File: packages/appeal-reply-service-api/src/models/blankModel.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
+const Mongoose = require('mongoose');
+const replySchema = require('./replySchema');
+
+const Model = Mongoose.model('Model', replySchema);
+const blankModel = new Model({
```
Comment (MrSimonEmms) [inline packages/appeal-reply-service-api/src/models/blankModel.js:5]: I presume you're going to be changing this later?

**Review by MrSimonEmms — CHANGES_REQUESTED** [review]
Need to get docker-compose, Helm and GitHub Actions configured (will speak via Slack)

Also, this feels like this should be merged into a `feature` branch before going into `master`. I'm not against putting this into `master` like this as this looks to be a working copy, but `master` is for stuff that can be released

File: packages/appeal-reply-service-api/api/openapi.yaml (hunk 1 above)
Comment (swdpcomputing) [reply packages/appeal-reply-service-api/api/openapi.yaml:11]: Done

File: packages/appeal-reply-service-api/data/README.md (hunk 2 above)
Comment (swdpcomputing) [reply packages/appeal-reply-service-api/data/README.md:1]: Weird, already did that. Deleted.

File: packages/appeal-reply-service-api/data/lpa-trialists.json (hunk 3 above)
Comment (swdpcomputing) [reply packages/appeal-reply-service-api/data/lpa-trialists.json:1]: Deleted

File: packages/appeal-reply-service-api/globalConfig.json (hunk 4 above)
Comment (swdpcomputing) [reply packages/appeal-reply-service-api/globalConfig.json:1]: Done.

File: packages/appeal-reply-service-api/src/models/blankModel.js (hunk 5 above)
Comment (swdpcomputing) [reply packages/appeal-reply-service-api/src/models/blankModel.js:5]: Depending on how we test.

File: packages/appeal-reply-service-api/globalConfig.json (hunk 4 above)
Comment (MrSimonEmms) [reply packages/appeal-reply-service-api/globalConfig.json:1]: Can you delete this file as well?

**Review by MrSimonEmms — CHANGES_REQUESTED** [review]
Looks pretty good.

We need to check that the pipeline works - by removing the `globalConfig.json` file, that should trigger it.

Also, what thoughts do you have about putting this into a `feature` branch rather than into `master`? Normally, we create the `feature` branch with the feature ticket (not sure if you're doing that - not the end of the world if you're not)

Can you also also raise a ticket on your board and assign to me to sort out the Helm deployment and I'll do that tomorrow

File: .github/workflows/appeal-reply-service-api.yml [hunk 6]
```diff
@@ -0,0 +1,110 @@
+name: Appeal Reply Service API
+
+env:
+  APP_DIR: appeals-reply-service-api
```
Comment (MrSimonEmms) [inline .github/workflows/appeal-reply-service-api.yml:4]: Change all instances of `appeals-reply-service-api` to `appeal-reply-service-api`

**Review by MrSimonEmms — CHANGES_REQUESTED** [review]

**Comment by davehlewis** [comment]
Changed the base branch to the right feature after our discussion

File: .github/workflows/appeal-reply-service-api.yml (hunk 6 above)
Comment (swdpcomputing) [reply .github/workflows/appeal-reply-service-api.yml]: Done.

**Review by MrSimonEmms — APPROVED** [review]

**Pull request merged**